File: app/src/main/java/ru/embersoft/favoritelist/Adapters/FavAdapter.java
package ru.embersoft.favoritelist.Adapters;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import ru.embersoft.favoritelist.Helpers.FavDB;
// import ru.embersoft.favoritelist.Helpers.FavItem;
// import ru.embersoft.favoritelist.R;

public class FavAdapter extends RecyclerView.Adapter<FavAdapter.ViewHolder> {

    private Context context;
    //private List<FavItem> favItemList;
    private FavDB favDB;
    //private DatabaseReference refLike;

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
       // View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.fav_item,
        //        parent, false);
       // favDB = new FavDB(context);
       // return new ViewHolder(view);
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }


    public class ViewHolder extends RecyclerView.ViewHolder {

        TextView favTextView;
        Button favBtn;
        ImageView favImageView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

        }
    }

}

File: app/src/main/java/com/example/mr_framer_grocer/Adapter/FavAdapter.kt
package com.example.mr_framer_grocer.Adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.mr_framer_grocer.Database.favRoom.Fav
import com.example.mr_framer_grocer.ProductDetailsActivity
import com.example.mr_framer_grocer.R

class FavAdapter(internal var context: Context, internal var itemList: MutableList<Fav>):
    RecyclerView.Adapter<MyFavViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyFavViewHolder {
        val itemView = LayoutInflater.from(context).inflate(R.layout.activity_fav_layout, parent, false)

        return MyFavViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyFavViewHolder, position: Int) {
        Glide.with(context).load(itemList[position].image).into(holder.fav_image)
        holder.fav_name.text = itemList[position].name
        holder.fav_price.text = context.getString(R.string.price, itemList[position].price)

        // view item details
        holder.fav_image.setOnClickListener {
            // if any of the product is clicked, direct to product details page
            val intent = Intent(context, ProductDetailsActivity::class.java)
                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)

            intent.putExtra("id", itemList[position].id.toString())
            intent.putExtra("name", itemList[position].name)
            intent.putExtra("price", itemList[position].price)
            intent.putExtra("weight", itemList[position].weight)
            intent.putExtra("img", itemList[position].image)
            intent.putExtra("category", itemList[position].category)
            intent.putExtra("stock", itemList[position].stock)
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return itemList.size
    }
}